examples/lines_bars_and_markers/scatter_masked.py

```diff
@@ -3,30 +3,77 @@
 Scatter Masked
 ==============
 
-Mask some data points and add a line demarking
-masked regions.
+A NumPy masked array (see `numpy.ma`) can be passed to `.Axes.scatter` or
+`.pyplot.scatter` as the value of the *s* parameter in order to exclude certain
+data points from the plot.
+
+This example uses this technique so that data points within a particular radius
+are not included in the plot.
 
 """
 import matplotlib.pyplot as plt
 import numpy as np
 
-# Fixing random state for reproducibility
+# Fix random state for reproducibility
 np.random.seed(19680801)
 
-
+# Create N random (x, y) points
 N = 100
-r0 = 0.6
 x = 0.9 * np.random.rand(N)
 y = 0.9 * np.random.rand(N)
-area = (20 * np.random.rand(N))**2  # 0 to 10 point radii
-c = np.sqrt(area)
-r = np.sqrt(x ** 2 + y ** 2)
-area1 = np.ma.masked_where(r < r0, area)
-area2 = np.ma.masked_where(r >= r0, area)
-plt.scatter(x, y, s=area1, marker='^', c=c)
-plt.scatter(x, y, s=area2, marker='o', c=c)
-# Show the boundary between the regions:
-theta = np.arange(0, np.pi / 2, 0.01)
-plt.plot(r0 * np.cos(theta), r0 * np.sin(theta))
+
+# Create masked size array based on calculation of x and y values
+size = np.full(N, 36)
+radius = 0.6
+masked_size = np.ma.masked_where(radius > np.sqrt(x ** 2 + y ** 2), size)
+
+# Plot data points using masked array
+subplot_kw = {
+    'aspect': 'equal',
+    'xlim': (0, max(x)),
+    'ylim': (0, max(y))
+}
+fig, ax = plt.subplots(subplot_kw=subplot_kw)
+ax.scatter(x, y, s=masked_size, marker='^', c="mediumseagreen")
+
+# Show the boundary between the regions
+theta = np.arange(-0, np.pi * 2, 0.01)
+circle_x = radius * np.cos(theta)
+circle_y = radius * np.sin(theta)
+ax.plot(circle_x, circle_y, c="black")
 
 plt.show()
+
+###############################################################################
+# This technique can also be used to plot a decision boundary, rather than
```

Comment:
Ah, so for context, this is the specific term for what the original example was demonstrating.  I was trying to add the keyword to make it more discoverable, since I think this is something that people are likely to search for

I moved it to a separate plot in order to make a simpler example that highlights just the `np.ma` aspect, because a decision boundary is a good amount more complicated than the minimum necessary code to demonstrate masking

What is the norm for this kind of domain-specific example? Some ideas I have are:

1. I could **add an explanation** of what a decision boundary is as well as the conventional way of showing it in graph form, but that would make the text a lot longer. And it looks like most of the graphs in the example gallery don't have extended text explanations
2. I could **link out to an external resource**. I see some other example gallery pages that use Wikipedia links but the [page about decision boundaries](https://en.wikipedia.org/wiki/Decision_boundary) is pretty short and doesn't include the relevant visual. Are there other standard sites to reference?
3. I could **leave this graph off of the gallery example altogether**, and start working on a tutorial page specifically about plotting decision boundaries

What do you recommend?